packages/shared/scope-hoisting/src/generate.js

```diff
@@ -1,44 +1,78 @@
 // @flow
 
-import type {Bundle, BundleGraph} from '@parcel/types';
-import type {File} from '@babel/types';
+import type {Bundle, BundleGraph, PluginOptions} from '@parcel/types';
+import type {
+  ExpressionStatement,
+  File,
+  Statement,
+  StringLiteral,
+} from '@babel/types';
 
 import babelGenerate from '@babel/generator';
 import nullthrows from 'nullthrows';
 import {isEntry} from './utils';
+import SourceMap from '@parcel/source-map';
+import * as t from '@babel/types';
+import template from '@babel/template';
 
-export function generate(bundleGraph: BundleGraph, bundle: Bundle, ast: File) {
-  let {code} = babelGenerate(ast, {
-    minified: bundle.env.minify,
-    comments: true, // retain /*@__PURE__*/ comments for terser
-  });
+const REGISTER_TEMPLATE = template.statement<
+  {|ID: StringLiteral, STATEMENTS: Array<Statement>|},
+  ExpressionStatement,
+>('parcelRequire.registerBundle(ID, function () { STATEMENTS; })');
+const WRAPPER_TEMPLATE = template.statement<
+  {|STATEMENTS: Array<Statement>|},
+  ExpressionStatement,
+>('(function () { STATEMENTS; })()');
 
+export function generate(
+  bundleGraph: BundleGraph,
+  bundle: Bundle,
+  ast: File,
+  options: PluginOptions,
+) {
   // $FlowFixMe
```

Comment:
We can actually remove the fixme by wrapping `...meta.interpreter` in `assertString` (from `./utils`)